hazelcast/src/main/java/com/hazelcast/jet/core/DAG.java

```diff
@@ -312,6 +308,32 @@ public Iterator<Vertex> iterator() {
         return validate().iterator();
     }
 
+    public void markAsPrunable() {
+        this.dagMembersMarkedAsPrunable = true;
+    }
+
+    public boolean areDagMembersMarkedAsPrunable() {
+        return dagMembersMarkedAsPrunable;
+    }
+
+    /**
+     * Returns true, if members in current DAG are allowed
+     * to be pruned without additional conditions.
+     */
+    public boolean membersArePrunable() {
+        if (!dagMembersMarkedAsPrunable) {
+            return false;
+        }
+
+        // Check each meta-supplier if it is possible to eliminate processor creation.
+        for (Vertex v : this) {
+            if (v.getMetaSupplier().doesWorkWithoutInput()) {
```

Comment:
if we remove entire member, we can remove processor instances regardless if they do work without input or not. There should be any difference in outcome of the job when there is different number of members (topology is variable any way) provided that the job executes on >= 1 member (sic! ofc. execution on 0 members could produce different result)

Reply:
> if we remove entire member, we can remove processor instances regardless if they do work without input or not

From my perspective, it is reverse dependency -- we remove an entire member only if processor instances work without input.

Reply:
we definitely need to discuss that